dbt/adapters/glue/impl.py

```diff
@@ -821,12 +828,45 @@ def delta_create_table(self, target_relation, request, primary_key, partition_ke
                 logger.error(e)
             self._update_additional_location(target_relation, location)
 
+    def _strip_catalog_from_schema(self, schema):
+        """
+        Utility function to remove the iceberg catalog name from schema if present.
+
+        Args:
+            schema (str): The schema name, possibly prefixed with catalog name
+
+        Returns:
+            str: The schema name without catalog prefix
+        """
+        iceberg_catalog = self.get_custom_iceberg_catalog_namespace()
+
+        # If no custom catalog is configured, return schema as is
+        if not iceberg_catalog:
+            return schema
+
+        # If schema starts with catalog name followed by a dot, remove it
+        if schema.startswith(f"{iceberg_catalog}."):
+            return schema[len(iceberg_catalog) + 1:]
+
+        # If schema contains catalog name with dots on both sides (e.g., in middle)
+        # This handles cases where catalog might be in the middle of a schema path
```

Comment:
I thought Iceberg catalog name is in the prefix (if exists), not in the middle. Is there any use case to set the catalog name is in the middle?

Reply:
Simplied